app/service/auth_svc.py

```diff
@@ -123,14 +128,35 @@ async def _check_credentials(user_map, username, password):
     async def _ldap_login(self, username, password):
         server = ldap3.Server(self.ldap_config.get('server'))
         dn = self.ldap_config.get('dn')
-        userattr = self.ldap_config.get('userattr') or 'uid'
-        userstring = '{}={},{}'.format(userattr, username, dn)
-        with ldap3.Connection(server, user=userstring, password=password) as conn:
+        user_attr = self.ldap_config.get('user_attr') or 'uid'
+        user_string = '{}={},{}'.format(user_attr, username, dn)
+
+        with ldap3.Connection(server, user=user_string, password=password) as conn:
             if conn.bind():
+                if username not in self.user_map:
+                    group = await self._ldap_get_group(conn, dn, username, user_attr)
+                    await self.create_user(username, None, group)
                 return True
             else:
                 return False
 
+    async def _ldap_get_group(self, connection, dn, username, user_attr):
+        group_attr = self.ldap_config.get('group_attr') or 'objectClass'
+        red_group_name = self.ldap_config.get('red_group') or 'red'
+
+        try:
+            connection.search(dn, '({}={})'.format(user_attr, username), attributes=[group_attr])
+        except LDAPAttributeError:
+            self.log.error('Invalid group_attr in config: {}'.format(group_attr))
+            return 'blue'
+
+        groups_result = connection.entries[0][group_attr].value
+        if ((isinstance(groups_result, list) and red_group_name in groups_result)
+                or red_group_name == groups_result):
+            return 'red'
```

Comment:
Return the Access enum instead of hardcoded string

Reply:
This was done to match how users were already being created in the existing code. `apply()` creates `User`s with groups named after the config file. The default config file uses groups called `red` and `blue`. Using the access enum gives groups of `1` and `2`, which leads to errors. 

Do you want me to return the enum here and convert to string elsewhere in the call stack, like `create_user`?